index.js

```diff
@@ -129,6 +152,41 @@
       Object.prototype.toString.call(x).slice('[object '.length, -']'.length);
   }
 
-  return type;
+  //# parse :: String -> { namespace :: Nullable String, name :: String, version :: Number }
+  //.
+  //. Takes any String and parses it according to the [specification][3],
+  //. returning an object with a `namespace`, a `name` and a `version` field.
+  //.
+  //. ```javascript
+  //. > parse('my-package/List@2');
+  //. {namespace: 'my-package', name: 'List', version: 2}
+  //.
+  //. > parse('nonsense!');
+  //. {namespace: null, name: 'nonsense!', version: 0}
+  //.
+  //. > function Identity(x) { this.value = x; }
+  //. > Identity['@@type'] = 'my-package/Identity@1';
+  //. > parse(type(new Identity(0)));
+  //. {namespace: 'my-package', name: 'Identity', version: 1}
+  //. ```
+  function parse(s) {
+    var parsed = RPARSE.exec(s);
+    if (parsed == null) return TypeIdentifier(null, s, 0);
+    return TypeIdentifier(
+      parsed[1] == null ? null : parsed[1],
+      parsed[2],
+      parsed[3] == null ? 0 : Number(parsed[3])
+    );
```

Comment:
I find `TypeIdentifier` to be an awkward function. I have thought of a way to remove it which may or may not be sensible. What do you think of the following?

```javascript
var groups = RPARSE.exec(s);
return {
  namespace: groups == null || groups[1] == null ? null : groups[1],
  name:      groups == null                      ? s    : groups[2],
  version:   groups == null || groups[3] == null ? 0    : Number(groups[3])
};
```

I like the fact that default values (`null` and `0`) are specified in one place rather than two. Also, the length of the longest line is 79 characters. ;)